components/sup/src/manager/mod.rs

```diff
@@ -882,6 +896,22 @@ impl Manager {
         Ok(())
     }
 
+    fn update_peers_from_watch_file(&mut self) -> Result<()> {
+        if !self.butterfly.need_peer_seeding() {
+            return Ok(());
+        }
+        match self.peer_watcher {
+            None => Ok(()),
+            Some(ref watcher) => {
+                if watcher.has_fs_events() {
+                    let members = watcher.get_members()?;
+                    self.butterfly.member_list.set_initial_members(members);
```

Comment:
When the peer file changes, how do the new peers get added into the list of "active" members? I don't see where the peer list ever gets read again after the initial startup.

Reply:
The `--peer-watch-file` flag works a bit like the `--peer` flag - it sets the initial members list and then the butterfly outbound thread will ping each member from that list. I guess that the inbound thread will handle the pong reply and initiate the move of the member from the "initial" list to the "active" list.

Also, like in the `--peer` flag, once we are a part of the ring, we don't really care about this anymore.